File: src/Model/MainModelImpl.php
<?php

namespace Lab_Web\Model;

class MainModelImpl implements MainModel {

    private $doFrontendTimeUpdate = true;
    private $startTime;

    public function __construct($startTime = null) {
        if ($startTime == null) {
            $startTime = microtime(true);
        }

        $this->startTime = $startTime;
    }

    public function getCurrentTime() {
        return time();
    }

    public function getElapsedTime() {
        return microtime(true) - $this->startTime;
    }

    public function doFrontendTimeUpdate() {
        return $this->doFrontendTimeUpdate;
    }

    /**
     * @param bool $doFrontendTimeUpdate
     */
    public function setDoFrontendTimeUpdate($doFrontendTimeUpdate) {
        $this->doFrontendTimeUpdate = $doFrontendTimeUpdate;
    }
}

File: src/ru/byprogminer/Lab2_Web/model/MainModelImpl.java
package ru.byprogminer.Lab2_Web.model;

import java.time.ZonedDateTime;

public class MainModelImpl implements MainModel {

    private boolean doFrontendTimeUpdate = true;
    private final long startTime;

    public MainModelImpl(Long startTime) {
        if (startTime == null) {
            startTime = System.nanoTime();
        }

        this.startTime = startTime;
    }

    @Override
    public ZonedDateTime getCurrentTime() {
        return ZonedDateTime.now();
    }

    @Override
    public double getElapsedTime() {
        return ((double) (System.nanoTime() - startTime)) / 10000000;
    }

    @Override
    public boolean doFrontendTimeUpdate() {
        return doFrontendTimeUpdate;
    }

    public void setDoFrontendTimeUpdate(boolean doFrontendTimeUpdate) {
        this.doFrontendTimeUpdate = doFrontendTimeUpdate;
    }
}
